src/utils.rs

```diff
@@ -20,10 +22,34 @@ pub fn time_as_num(time: SystemTime) -> f64 {
     (time.duration_since(UNIX_EPOCH).unwrap().as_millis() as f64) / 1000.0
 }
 
+/// Write a byte slice to disk.
+///
+/// Succeeds if the file already exists with identical contents.
+/// On the other hand, an AlreadyExists error is returned if the file exists with different contents.
+pub fn write_file_idempotent(path: &Path, contents: &[u8]) -> io::Result<()> {
```

Comment:
I don't love this name, but I also couldn't find a better one that expresses the idea of detecting conflicts.